gslab_scons/tests/_side_effects.py

```diff
@@ -0,0 +1,274 @@
+        source = match.group('source')
+        log    = '%s.Rout' % re.sub('\.R', '', source)
+
+    if executable == "R CMD BATCH" and log:
+        with open(log.strip(), 'wb') as log_file:
+            log_file.write('Test log\n')
+
+
+def python_side_effect(*args, **kwargs):
+    command = args[0]
+    match   = helpers.command_match(command, 'python')
+
+    if match.group('log'):
+        log_path = re.sub('(\s|>)', '', match.group('log'))
+        with open(log_path, 'wb') as log_file:
+            log_file.write('Test log')
+
+
+def matlab_side_effect(*args, **kwargs):
+    '''Mock os.system for Matlab commands'''
+    try:
+        command = kwargs['command']
+    except KeyError:
+        command = args[0]
+
+    log_match = re.search('> (?P<log>[-\.\w\/]+)', command)
+
+    if log_match:
+        log_path = log_match.group('log')
+        with open(log_path, 'wb') as log_file:
+            log_file.write('Test log')
+
+    return None
+
+
+def make_stata_side_effect(recognised_command):
+    '''
+    Make a side effect mocking the behaviour of 
+    subprocess.check_output() when `recognised_command` is
+    the only recognised system command. 
+    '''
+    def stata_side_effect(*args, **kwargs):
+        command = args[0]
+        match   = helpers.command_match(command, 'stata')
+
+        if match.group('executable') == recognised_command:
+            # Find the Stata script's name
+            script_name = match.group('source')
+            stata_log   = os.path.basename(script_name).replace('.do','.log')
+            
+            # Write a log
+            with open(stata_log, 'wb') as logfile:
+                logfile.write('Test Stata log.\n')
+
+        else:
+            # Raise an error if the exectuable is not the only recognised one.
+            raise subprocess.CalledProcessError(1, command)
+
+    return stata_side_effect
+
+
+def lyx_side_effect(*args, **kwargs):
+    '''
+    This side effect mocks the behaviour of a system call.
+    The mocked machine has lyx set up as a command-line executable
+    and can export .lyx files to .pdf files only using 
+    the "-e pdf2" option.
+    '''
+    # Get and parse the command passed to os.system()
+    command = args[0]
+    match = re.match('\s*'
+                        '(?P<executable>\w+)'
+                        '\s+'
+                        '(?P<option>-\w+ \w+)?'
+                        '\s*'
+                        '(?P<source>[\.\/\w]+\.\w+)?'
+                        '\s*'
+                        '(?P<log_redirect>\> [\.\/\w]+\.\w+)?',
+                     command)
+
+    executable   = match.group('executable')
+    option       = match.group('option')
+    source       = match.group('source')
+    log_redirect = match.group('log_redirect')
+
+    option_type    = re.findall('^(-\w+)',  option)[0]
+    option_setting = re.findall('\s(\w+)$', option)[0]
+
+    is_lyx = bool(re.search('^lyx$', executable, flags = re.I))
```

Comment:
They aren't on my computer. I'm not sure about Windows but it wouldn't be difficult for me to check this later today.